docs/introduction/product-search-via-elasticsearch.md

```diff
@@ -0,0 +1,58 @@
+# Product Searching via Elasticsearch
+To provide the best possible performance, frontend product searching and autocomplete
+leverages [Elasticsearch technology](https://www.elastic.co/products/elasticsearch).
+
+## Product data export
+Following product attributes are exported into Elasticsearch (i.e. the search is performed on these fields only):
+* name
+* catnum
+* partno
+* ean
+* description
+* short description
+
+Data of all products are exported into Elasticsearch by CRON module (`ElasticsearchExportCronModule`) once an hour.
+Alternatively, you can force the export manually using `elasticsearch-products-export` phing target.
+
+## Searching for products
+
+We use the same method for searching and for autocomplete, so results are always the same.
+
+Understanding elasticsearch searching is difficult.
+But if we simplify, we can say that the search term is searched in attributes and is prioritized in following order:
+* ean - exact match
+* name - match any of words
+* name - match any of words ignoring diacritics
+* catnum - exact match
+* name - match any of words in root form
+* partno - exact match
+* name - match in first couple of letters of any word
+* name - match in first couple of letters of any word ignoring diacritics
+* ean - match in first couple of letters
+* catnum - match in first couple of letters
+* partno - match in first couple of letters
+* short description - match anywhere
+* description - match anywhere
```

Comment:
So if I will search "ABC", the products with "ABC" in a name will be higher than product witch exact match in catnum? I'm not sure what is the purpose of this article. Is the purpose just showing that Shopsys Framework has elastic? If so, then it is fulfilled. Or is the purpose to show to the users how they can use our elastic and what will be the results. If so, then I don't see this article as a fulfilling of this purpose :/

Reply:
Purpose of this article is to shortly say "We use elasticsearch, these data are exported, searching is done in this way, indexes definition is here and you have to do this to make your elasticsearch running".

Honestly we didn't spent much time on the documentation for two reasons:
* Making good documentation will take a lot of time
* Elasticsearch usage will be moved into microservice

Reply:
"Making good documentation will take a lot of time" - I disagree. "Elasticsearch usage will be moved into microservice" this reason means that description of elastic will be the part of implementation of microservice? I was expecting that the docs will be done and just some minor changes will be needed after move into microservice.

Reply:
Hi, I have added fixup commit, I hope now it is better :wink: